### ISSUE TITLE
More detailed error reporting

### ISSUE THREAD

**Issue by epage** [steveklabnik/dir-diff#11]
I've got my own [home grown tool for diffing directories](https://github.com/cobalt-org/cobalt.rs/blob/master/tests/mod.rs#L29) but I'd much have a reusable crate.

API Ideas:
- Enum of `LeftExists`, `RightExists`, `BothExist(Option<Difference>)`
- `diff_dir_list(path, path)` provides an iterator of `(path, enum(None))`
  - doesn't do content diffing allowing people to do custom solutions for #6
- `diff_dir(path, path)` provides an iterator of `(path, enum)`
  - Take advantage of `difference` like [`assert_cli`](
- `assert_dir_eq!`
  - provides a default policy for failing (whether it be `panic` on first failure or summarize the all differences)

**Comment by steveklabnik** [comment]
Hey @epage ! I'm not really working on dir-diff these days, but if you want to hack on it, I'd be happy to give you commit rights, etc 👍

**Comment by epage** [comment]
That works.

Any vision or implementation goals you had that you'd like me to maintain with my changes?

**Comment by steveklabnik** [comment]
Not specifically; I had originally created this library as a way of doing testing; you could have a fixture with the directories you planned on creating, run the tests, and diff them. I'm open to whatever though. Adding you as a collaborator right now!

**Comment by savente93** [comment]
@epage Don't mean to be pushy, but it seems like there hasn't been much work on this for a while. If you want I'd be happy to do some work on it, but there are several PRs in a half finished state so I'm not sure what would be a good starting point. Got anything you'd like me to work on? I'd be happy to either try and work on the open PRs or start something in parallel, just let me know what you'd prefer.

**Comment by epage** [comment]
@savente93 what is it you are wanting to accomplish with dir-diff?  Knowing that would be a good start.

**Comment by savente93** [comment]
I'm working on a (new) tool to extract python documentation and render it out as markdown. What specifically for `dir-diff` I have a few integration tests and the reference output in the repository so I just want to make sure that the output and the reference answer are the same. 

That use case is technically already covered but as you detail above at the very least I'd want reporting about what exists in one but not the other, and to get the output of the file diffs if they exist but are different. So far, that's all I need, I don't need to handle the diff in any way just yet. 


 I have some code in my repo for it, so it's not an urgent need, but for me it's more boiler plate so if I can clean it up and contribute it to a dependency that wants to do it that's all the better. So I'm mostly just checking if I can help out, more than that I actually need something.
